common/Root.tsx

```diff
@@ -50,16 +50,10 @@ export default class Root extends Component<Props, State> {
     const routes = (
       <Switch>
         <Route exact={true} path="/" component={GenerateWallet} />
-        <Route path="/generate" component={GenerateWallet}>
-          <Route path="keystore" component={GenerateWallet} />
-          <Route path="mnemonic" component={GenerateWallet} />
-        </Route>
+        <Route path="/generate" component={GenerateWallet} />
```

Comment:
Upon testing, I've found that what this means is that this matches _every_ `/generate` route. I think what would be preferrable is to make all of these routes `exact={true}`, and to specify all routes in here:

```
<Route exact={true} path="/generate" component={GenerateWallet} />
<Route exact={true} path="/generate/keystore" component={GenerateWallet} />
<Route exact={true} path="/generate/mnemonic" component={GenerateWallet} />
```

Right now if you go to `/generate/likewhateverman`, you don't get the 404 page, you just get a contentless template page.

Reply:
Just another point for being explicit about routes, it would allow us to do something like this: https://www.npmjs.com/package/react-static-webpack-plugin and get rid of HashRouter, by having actual HTML files for each route. Might also simplify our gh-pages configuration as well so that non-index pages aren't 404ing!

Reply:
I've added `<RouteWithSubRoutes />` component which should now take care of 404 pages  without having to use `exact={true}` on every path.

Also I don't think we're using any dynamic routes, so something like [react-static-webpack-plugin](npmjs.com/package/react-static-webpack-plugin) should still work